After migration on flutter 1.17 application wont build

> Compiler message:
../flutter-sdk/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:168:13: Error: Expected 0 type arguments.
  Map<Type, Action<Intent>> _actionMap;
            ^
../flutter-sdk/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:178:23: Error: Type 'ActivateIntent' not found.
  void _actionHandler(ActivateIntent intent){
                      ^^^^^^^^^^^^^^
../flutter-sdk/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:173:25: Error: Expected 0 type arguments.
    _actionMap = <Type, Action<Intent>>{
                        ^^^^^^
../flutter-sdk/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:174:38: Error: 'ActivateIntent' isn't a type.
      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
                                     ^^^^^^^^^^^^^^
../flutter-sdk/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:174:53: Error: Too few positional arguments: 1 required, 0 given.
      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
                                                    ^
../flutter-sdk/packages/flutter/lib/src/widgets/actions.dart:128:9: Context: Found this candidate, but the arguments don't match.
  const CallbackAction(LocalKey intentKey, {@required this.onInvoke})
        ^^^^^^^^^^^^^^
../flutter-sdk/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:174:7: Error: The getter 'ActivateIntent' isn't defined for the class '_CircularCheckBoxState'.
 - '_CircularCheckBoxState' is from 'package:circular_check_box/circular_check_box.dart' ('../flutter-sdk/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'ActivateIntent'.
      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
      ^^^^^^^^^^^^^^
../flutter-sdk/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:173:40: Error: A value of type 'Map<Type, Action>' can't be assigned to a variable of type 'Map<Type, invalid-type>'.
 - 'Map' is from 'dart:core'.
 - 'Type' is from 'dart:core'.
 - 'Action' is from 'package:flutter/src/widgets/actions.dart' ('../flutter-sdk/packages/flutter/lib/src/widgets/actions.dart').
    _actionMap = <Type, Action<Intent>>{
                                       ^
../flutter-sdk/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:178:23: Error: 'ActivateIntent' isn't a type.
  void _actionHandler(ActivateIntent intent){
                      ^^^^^^^^^^^^^^
../flutter-sdk/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:226:16: Error: The argument type 'Map<Type, invalid-type>' can't be assigned to the parameter type 'Map<LocalKey, Action Function()>'.
 - 'Map' is from 'dart:core'.
 - 'Type' is from 'dart:core'.
 - 'LocalKey' is from 'package:flutter/src/foundation/key.dart' ('../flutter-sdk/packages/flutter/lib/src/foundation/key.dart').
 - 'Action' is from 'package:flutter/src/widgets/actions.dart' ('../flutter-sdk/packages/flutter/lib/src/widgets/actions.dart').
      actions: _actionMap,
               ^
Target kernel_snapshot failed: Exception: Errors during snapshot creation: null

Launching lib\main.dart on roarV95 in debug mode...
Running Gradle task 'assembleDebug'...

Compiler message:
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:168:13: Error: Expected 0 type arguments.
  Map<Type, Action<Intent>> _actionMap;
            ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:178:23: Error: Type 'ActivateIntent' not found.
  void _actionHandler(ActivateIntent intent){
                      ^^^^^^^^^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:173:25: Error: Expected 0 type arguments.
    _actionMap = <Type, Action<Intent>>{
                        ^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:174:38: Error: 'ActivateIntent' isn't a type.
      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
                                     ^^^^^^^^^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:174:53: Error: Too few positional arguments: 1 required, 0 given.
      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
                                                    ^
/C:/src/flutter/packages/flutter/lib/src/widgets/actions.dart:128:9: Context: Found this candidate, but the arguments don't match.
  const CallbackAction(LocalKey intentKey, {@required this.onInvoke})
        ^^^^^^^^^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:174:7: Error: The getter 'ActivateIntent' isn't defined for the class '_CircularCheckBoxState'.
 - '_CircularCheckBoxState' is from 'package:circular_check_box/circular_check_box.dart' ('/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'ActivateIntent'.
      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
      ^^^^^^^^^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:173:40: Error: A value of type 'Map<Type, Action>' can't be assigned to a variable of type 'Map<Type, invalid-type>'.
 - 'Map' is from 'dart:core'.
 - 'Type' is from 'dart:core'.
 - 'Action' is from 'package:flutter/src/widgets/actions.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/actions.dart').
    _actionMap = <Type, Action<Intent>>{
                                       ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:178:23: Error: 'ActivateIntent' isn't a type.
  void _actionHandler(ActivateIntent intent){
                      ^^^^^^^^^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:226:16: Error: The argument type 'Map<Type, invalid-type>' can't be assigned to the parameter type 'Map<LocalKey, Action Function()>'.
 - 'Map' is from 'dart:core'.
 - 'Type' is from 'dart:core'.
 - 'LocalKey' is from 'package:flutter/src/foundation/key.dart' ('/C:/src/flutter/packages/flutter/lib/src/foundation/key.dart').
 - 'Action' is from 'package:flutter/src/widgets/actions.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/actions.dart').
      actions: _actionMap,
               ^

Compiler message:
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:168:13: Error: Expected 0 type arguments.
  Map<Type, Action<Intent>> _actionMap;
            ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:178:23: Error: Type 'ActivateIntent' not found.
  void _actionHandler(ActivateIntent intent){
                      ^^^^^^^^^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:173:25: Error: Expected 0 type arguments.
    _actionMap = <Type, Action<Intent>>{
                        ^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:174:38: Error: 'ActivateIntent' isn't a type.
      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
                                     ^^^^^^^^^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:174:53: Error: Too few positional arguments: 1 required, 0 given.
      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
                                                    ^
/C:/src/flutter/packages/flutter/lib/src/widgets/actions.dart:128:9: Context: Found this candidate, but the arguments don't match.
  const CallbackAction(LocalKey intentKey, {@required this.onInvoke})
        ^^^^^^^^^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:174:7: Error: The getter 'ActivateIntent' isn't defined for the class '_CircularCheckBoxState'.
 - '_CircularCheckBoxState' is from 'package:circular_check_box/circular_check_box.dart' ('/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'ActivateIntent'.
      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
      ^^^^^^^^^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:173:40: Error: A value of type 'Map<Type, Action>' can't be assigned to a variable of type 'Map<Type, invalid-type>'.
 - 'Map' is from 'dart:core'.
 - 'Type' is from 'dart:core'.
 - 'Action' is from 'package:flutter/src/widgets/actions.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/actions.dart').
    _actionMap = <Type, Action<Intent>>{
                                       ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:178:23: Error: 'ActivateIntent' isn't a type.
  void _actionHandler(ActivateIntent intent){
                      ^^^^^^^^^^^^^^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/circular_check_box-1.0.2/lib/circular_check_box.dart:226:16: Error: The argument type 'Map<Type, invalid-type>' can't be assigned to the parameter type 'Map<LocalKey, Action Function()>'.
 - 'Map' is from 'dart:core'.
 - 'Type' is from 'dart:core'.
 - 'LocalKey' is from 'package:flutter/src/foundation/key.dart' ('/C:/src/flutter/packages/flutter/lib/src/foundation/key.dart').
 - 'Action' is from 'package:flutter/src/widgets/actions.dart' ('/C:/src/flutter/packages/flutter/lib/src/widgets/actions.dart').
      actions: _actionMap,
               ^
Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
build failed.

FAILURE: Build failed with an exception.

* Where:
Script 'C:\src\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 882

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command 'C:\src\flutter\bin\flutter.bat'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 24s
Exception: Gradle task assembleDebug failed with exit code 1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,3 +21,7 @@ Add the use of FocusNode and autofocus using FocusableActionDetector
 Add the use of VisualDensity
 Change the way the circular border is drawn, using Paint's strokeWidth and style
 Added example
+
+## [1.0.3] - 05/11/2020
+
+Fixed compile-time error on new Flutter 1.17

--- a/lib/circular_check_box.dart
+++ b/lib/circular_check_box.dart
@@ -1,4 +1,4 @@
-// Copyright 2015 The Chromium Authors. All rights reserved.
+// Copyright 2014 The Flutter Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -17,11 +17,28 @@ import 'package:flutter/widgets.dart';
 /// his behaviour is the same except this draws a circular shape instead of
 /// a squared.
 ///
+/// The checkbox itself does not maintain any state. Instead, when the state of
+/// the checkbox changes, the widget calls the [onChanged] callback. Most
+/// widgets that use a checkbox will listen for the [onChanged] callback and
+/// rebuild the checkbox with a new [value] to update the visual appearance of
+/// the checkbox.
+///
+/// The checkbox can optionally display three values - true, false, and null -
+/// if [tristate] is true. When [value] is null a dash is displayed. By default
+/// [tristate] is false and the checkbox's [value] must be true or false.
+///
 /// Requires one of its ancestors to be a [Material] widget.
 ///
 /// See also:
 ///
 ///  * [Checkbox]
+///  * [CircularCheckBoxListTile], which combines this widget with a [ListTile] so that
+///    you can give the checkbox a label.
+///  * [Switch], a widget with semantics similar to [CircularCheckBox].
+///  * [Radio], for selecting among a set of explicit values.
+///  * [Slider], for selecting a value in a range.
+///  * <https://material.io/design/components/selection-controls.html#checkboxes>
+///  * <https://material.io/design/components/lists.html#types>
 class CircularCheckBox extends StatefulWidget {
   /// Creates a material design circular checkbox.
   ///
@@ -121,7 +138,7 @@ class CircularCheckBox extends StatefulWidget {
 
   /// If true the checkbox's [value] can be true, false, or null.
   ///
-  /// Checkbox displays a dash when its value is null.
+  /// CircularCheckBox displays a dash when its value is null.
   ///
   /// When a tri-state checkbox ([tristate] is true) is tapped, its [onChanged]
   /// callback will be applied to true if the current value is false, to null if
@@ -165,17 +182,17 @@ class CircularCheckBox extends StatefulWidget {
 
 class _CircularCheckBoxState extends State<CircularCheckBox> with TickerProviderStateMixin {
   bool get enabled => widget.onChanged != null;
-  Map<Type, Action<Intent>> _actionMap;
+  Map<LocalKey, ActionFactory> _actionMap;
 
   @override
   void initState() {
     super.initState();
-    _actionMap = <Type, Action<Intent>>{
-      ActivateIntent: CallbackAction<ActivateIntent>(onInvoke: _actionHandler),
+    _actionMap = <LocalKey, ActionFactory>{
+      ActivateAction.key: _createAction,
     };
   }
 
-  void _actionHandler(ActivateIntent intent){
+  void _actionHandler(FocusNode node, Intent intent) {
     if (widget.onChanged != null) {
       switch (widget.value) {
         case false:
@@ -189,8 +206,15 @@ class _CircularCheckBoxState extends State<CircularCheckBox> with TickerProvider
           break;
       }
     }
-    final RenderObject renderObject = context.findRenderObject();
+    final RenderObject renderObject = node.context.findRenderObject();
     renderObject.sendSemanticsEvent(const TapSemanticEvent());
+  }
+
+  Action _createAction() {
+    return CallbackAction(
+      ActivateAction.key,
+      onInvoke: _actionHandler,
+    );
   }
 
   bool _focused = false;

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -1,6 +1,6 @@
 name: circular_check_box
 description: Same as the regular checkbox, but circular!
-version: 1.0.2
+version: 1.0.3
 author: Mash Ibtesum <mashnoon33@gmail.com>
 homepage: http://mash.studio/
 repository: https://github.com/mashnoon33/circular_check_box.git

